diff --git a/src/Visitors.ml b/src/Visitors.ml
index 7139ed7..f28925f 100644
--- a/src/Visitors.ml
+++ b/src/Visitors.ml
@@ -1258,7 +1258,7 @@ let type_decls (decls : type_declaration list) : structure =
          also parameterized over the type variable ['self], with a constraint
          that this is the type of [self]. This trick allows us to omit the types
          of the virtual methods, even if these types include type variables. *)
-    dump X.concrete X.ancestors [ ty_self, Invariant ] pself X.name ::
+    dump X.concrete X.ancestors [ ty_self, (NoVariance, NoInjectivity) ] pself X.name ::
     floating "VISITORS.END" [] ::
     []
   )]
